torchao/csrc/metal/int4mv_kernel.mm

```diff
@@ -0,0 +1,50 @@
+#include "int4mv_kernel.h"
+#include "mps/MPSStream.h"
+#include "mps/OperationUtils.h"
+namespace torchao {
+
+void int4mv(const uint8_t * A, const uint8_t * B,  uint32_t groupSize, const uint8_t * scalesAndZeros, uint8_t * C, std::array<uint32_t, 4> sizes, std::array<uint64_t, 4> nbytes, std::string A_scalar_type) {
```

Comment:
I know this is draft but whenever you pdate, describe the args. Like nbytes seem to suggest size per "tensor" arg. In that case why use std::array? (also pass by ref)